src/command_modules/azure-cli-sql/tests/test_sql_commands.py

```diff
@@ -1009,6 +1010,68 @@ def test_sql_elastic_pools_mgmt(self, resource_group, resource_group_location, s
+    def test_sql_capabilities(self):
+        location = 'westus'
+
+        # New capabilities are added quite frequently and the state of each capability depends
+        # on your subscription. So it's not a good idea to make strict checks against exactly
+        # which capabilities are returned. The idea is to just check the overall structure.
+
+        # Get all capabilities
+        self.cmd('sql show-capabilities -l {}'.format(location),
+                 checks=[
+                     # All statuses (default, available, and visible) are shown
+                     JMESPathCheckGreaterThan(
+                         "length(supportedServerVersions[].supportedEditions[]"
+                         ".supportedServiceLevelObjectives[?status == 'Default'])", 0),
+                     JMESPathCheckGreaterThan(
+                         "length(supportedServerVersions[].supportedEditions[]"
+                         ".supportedServiceLevelObjectives[?status == 'Available'])", 0),
+                     JMESPathCheckGreaterThan(
+                         "length(supportedServerVersions[].supportedEditions[]"
+                         ".supportedServiceLevelObjectives[?status == 'Visible'])", 0),
+                     # Full depth is preserved
+                     JMESPathCheckGreaterThan(
+                         'length(supportedServerVersions[].supportedElasticPoolEditions[]'
+                         '.supportedElasticPoolDtus[].supportedPerDatabaseMaxDtus[]'
+                         '.supportedPerDatabaseMinDtus[])', 0)
+                ])
+
+        # Filter by various depths
+        self.cmd('sql show-capabilities -l {} --depth 0'.format(location),
+                 checks=[])
+        self.cmd('sql show-capabilities -l {} --depth 1'.format(location),
+                 checks=[])
+        self.cmd('sql show-capabilities -l {} --depth 2'.format(location),
+                 checks=[])
+        self.cmd('sql show-capabilities -l {} --depth 3'.format(location),
+                 checks=[])
+        self.cmd('sql show-capabilities -l {} --depth 4'.format(location),
+                 checks=[])
+
+        # Search for edition that exists for both db and pool
+        edition1 = 'Premium'
+        self.cmd('sql show-capabilities -l {} --edition {}'.format(location, edition1),
+                 checks=[])
+
+        # Search for edition that exists for db only
+        edition2 = 'DataWarehouse'
+        self.cmd('sql show-capabilities -l {} --edition {}'.format(location, edition2),
+                 checks=[])
+
+        # Search for service objective
+        service_objective = 'P2'
+        self.cmd('sql show-capabilities -l {} --service-objective {}'
+                 .format(location, service_objective),
+                 checks=[])
+
+        # Show default only
+        self.cmd('sql show-capabilities -l {} --service-objective {} --status Default'
+                 .format(location, service_objective),
+                 checks=[])
```

Comment:
Additional scenario to add: search for incompatible edition/service objective, eg. Premium // S2